src/platform/plugins/shared/data/public/actions/filters/create_filters_from_value_click.ts

```diff
@@ -129,6 +134,68 @@ export const createFilter = async (
   return filter;
 };
 
+export const createFilterESQL = async (
+  table: Pick<Datatable, 'rows' | 'columns'>,
+  columnIndex: number,
+  rowIndex: number
+) => {
+  if (
+    !table ||
+    !table.columns ||
+    !table.columns[columnIndex] ||
+    !table.columns[columnIndex].meta ||
+    !table.columns[columnIndex].meta.sourceParams?.sourceField
+  ) {
+    return;
+  }
+  const column = table.columns[columnIndex];
+  const { indexPattern, sourceField, operationType, interval } = table.columns[columnIndex].meta
+    .sourceParams as {
+    indexPattern: string;
+    sourceField: string;
+    operationType: string;
+    interval: number;
+  };
+
+  const value: any = rowIndex > -1 ? table.rows[rowIndex][column.id] : null;
+  if (value === null || value === undefined) {
+    return;
+  }
+
+  const filters: Filter[] = [];
+
+  if (operationType === 'date_histogram') {
+    filters.push(
+      buildSimpleNumberRangeFilter(
+        sourceField,
+        {
+          gte: moment(value).toISOString(),
+          lt: moment(value).add(interval).toISOString(),
+          format: 'strict_date_optional_time',
+        },
+        indexPattern,
+        value
+      )
+    );
+  } else if (operationType === 'histogram') {
+    filters.push(
+      buildSimpleNumberRangeFilter(
+        sourceField,
+        {
+          gte: value,
+          lt: value + interval,
+        },
+        indexPattern,
+        value
+      )
+    );
+  } else if (sourceField) {
+    filters.push(buildSimpleExistFilter(sourceField, indexPattern));
```

Comment:
Does this mean we're always going to create either a range or an exists filter when clicking on a value?

Reply:
this is what is currently supported, on aggconfigs we also do ip range filters, ip prefix, terms and multiterms filters, but this is not yet supported on esql level at the moment (or by this PR)

Reply:
Can you let me know how to get into a situation in this PR where something other than an exists filter should be created? I can't seem to set up any Lens configurations that actually send the ES|QL query instead of the aggs.

Reply:
No I think you wont find any Lukas, I was also trying to test this scenario

Reply:
@lukas go to advanced settings and set time zone to UTC
then time charts should also work (basic ones) and date range should be updated when filtering